yjit/src/core.rs

```diff
@@ -703,7 +742,7 @@ impl Block {
 
         // Wrap the block in a reference counted refcell
         // so that the block ownership can be shared
-        Rc::new(RefCell::new(block))
+        BlockRef(Rc::new(RefCell::new(block)))
```

Comment:
I think it would be nice to have a `BlockRef::new` constructor here. Will make an eventual transition to Box easier too.